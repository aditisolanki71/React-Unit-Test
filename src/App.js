import './App.css';
import Counter from "./components/Counter";
import Matchers from './components/Matchers';
import MockComponent from './components/MockComponents/MockComponent';
import SnapshotExample from './components/SnapshotExample';
import Todos from './components/Todos';
function App() {
  return (
    <div className="App">
      Learn React
      <Counter />
      <Matchers />
      <Todos />
      <MockComponent />
      <SnapshotExample />
    </div>
  );
}

export default App;
