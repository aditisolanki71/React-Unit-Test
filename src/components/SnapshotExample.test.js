import { render } from "@testing-library/react"
import { createRenderer } from "react-dom/test-utils"
import renderer from "react-test-renderer"
import App from "../App"
import Todos from "./Todos"
describe("snapshot testing", () => {
    test("snapshot for App component", () => {
        //convert component into json format nd save as snapshot
        //if snapshot already exist than it compares it and update it.if not then create new snapshot
        const renderedComponent = renderer.create(<App/>).toJSON()
        expect(renderedComponent).toMatchSnapshot();
    })

    test("snapshot for Todos component", () => {
        const renderedComponent = renderer.create(<Todos/>).toJSON()
        expect(renderedComponent).toMatchSnapshot();
    })
})
