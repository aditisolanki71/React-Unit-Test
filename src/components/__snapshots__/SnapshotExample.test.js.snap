// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for App component 1`] = `
<div
  className="App"
>
  Learn React
  <h3>
    Counter Example
  </h3>
  <button
    data-testid="increment"
    onClick={[Function]}
  >
    +
  </button>
  <span
    data-testid="counter"
  >
    0
  </span>
  <button
    data-testid="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <h3>
    Matchers Example
  </h3>
  <h3>
    Async code
  </h3>
  <div
    data-testid="fullname"
  >
    Full name is:- 
    Aditi Solanki
  </div>
  <h3>
    Todo List
  </h3>
  <p>
    Loading....
  </p>
  <h3>
    Mock Component Example
  </h3>
  <div
    id="component3"
  >
    Component3
  </div>
  <div
    id="component1"
  >
    Component1
  </div>
  <div
    id="component2"
  >
    Component2
  </div>
  <h3>
    Counter Example
  </h3>
  <a
    href="https://jsonplaceholder.typicode.com/users"
  >
    Fake data
  </a>
</div>
`;

exports[`snapshot testing snapshot for Todos component 1`] = `
Array [
  <h3>
    Async code
  </h3>,
  <div
    data-testid="fullname"
  >
    Full name is:- 
    Aditi Solanki
  </div>,
  <h3>
    Todo List
  </h3>,
  <p>
    Loading....
  </p>,
]
`;
